app/src/main/java/com/example/githubusers/presenter/UserProfilePresenter.java

```diff
@@ -0,0 +1,48 @@
+package com.example.githubusers.presenter;
+
+import android.util.Log;
+
+import com.example.githubusers.model.UserProfile;
+import com.example.githubusers.service.GithubService;
+import com.example.githubusers.view.UserDetailParentView;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+
+public class UserProfilePresenter {
+    /* package */ static final String TAG = "UserProfilePresenter";
+    /* package */ final UserDetailParentView userProfileView;
+    private GithubService githubService;
+
+    public UserProfilePresenter(UserDetailParentView view) {
+        this.userProfileView = view;
+
+        if (this.githubService == null) {
+            this.githubService = new GithubService();
```

Comment:
Like I once mentioned, the Presenter class should receive the service class as an argument when being constructed. This is to make it unit testable. Kindly fix that. Good Job 👍